### DISCUSSION TITLE
wmse/

### DISCUSSION THREAD

**Discussion by giscus** [rajashekar/blog_comments#2] [General]
# wmse/

Tensorflow has two separate functions to calculate MSE (Mean square error). For Loss - tf.keras.loss.MeanSquaredError() For Metrics - tf.keras.metrics.MeanSquaredError()

https://www.rajashekar.org/wmse/

**Comment by shlagbaum** [comment]
this is great, but how do you pass sample_weights to model.compile?

**Reply by gaebor** [reply]
I know of three methods:
 * model.compile(loss_weights=...)
 * model.fit(sample_weight=...)
 * or as the documentation says:
```This argument is not supported when x is a dataset, generator, or keras.utils.PyDataset instance, instead provide the sample_weights as the third element of x```
like `tf.data.Dataset.from_tensor_slices((x, y, sample_weight)).batch(self.batch_size)`
